src/plugins/GUITestBase/src/tests/regression_scenarios/GTTestsRegressionScenarios_7001_8000.cpp

```diff
@@ -1210,6 +1211,54 @@ GUI_TEST_CLASS_DEFINITION(test_7410) {
     GTUtilsProjectTreeView::checkItem(os, "test_7410.aln");
 }
 
+GUI_TEST_CLASS_DEFINITION(test_7413) {
+    // Check that the distribution is uniform by the Kolmogorov-Smirnov test.
+    // https://colab.research.google.com/drive/1-F4pAh-n0BMXeZczQY-te-UcEJeUSP8Y?usp=sharing
+    DNASequenceGeneratorDialogFillerModel model(sandBoxDir + "/test_7413_1.fa");
+    model.percentA = 99;
+    model.percentC = 1;
+    model.percentG = 0;
+    model.percentT = 0;
+    model.length = 10000;
+
+    auto checkUniformDistribution = [&model, &os]() {
+        GTUtilsDialog::waitForDialog(os, new DNASequenceGeneratorDialogFiller(os, model));
+        GTMenu::clickMainMenuItem(os, {"Tools", "Random sequence generator..."});
+
+        GTUtilsSequenceView::checkSequenceViewWindowIsActive(os);
+        QString sequence = GTUtilsSequenceView::getSequenceAsString(os);
+
+        QVector<int> numericalSequence(model.length);
+        for (int i = 0; i < model.length; i++) {
+            numericalSequence[i] = sequence[i] == 'C' ? 1 : 0;
+        }
+
+        QVector<int> empiricalSum = numericalSequence;
+        for (int i = 1; i < model.length; i++) {
+            empiricalSum[i] += empiricalSum[i - 1];
+        }
+
+        int sumNumSeq = std::accumulate(numericalSequence.constBegin(), numericalSequence.constEnd(), 0);
+        QVector<double> difference(model.length);
+        for (int i = 0; i < model.length; i++) {
+            difference[i] = std::abs((double(i) + 1) / model.length - double(empiricalSum[i]) / sumNumSeq);
```

Comment:
You do not need a 'difference' as an array here. All you need is a single int: maxDifference.